vdom/core.py

```diff
@@ -8,19 +8,34 @@
 This module provides the core implementation for the VDOM (Virtual DOM).
 
 """
-def toJSON(el):
-    """Convert an element to JSON"""
-    if(type(el) is str):
-        return el
+
+from jsonschema import validate, Draft4Validator, ValidationError
+from functools import partial
+
+def toJSON(el, schema=None):
+    """Convert an element to JSON
+
+    If you wish to validate the JSON, pass in a schema via the schema keyword argument.
+    If a schema is provided, this raises a ValidationError if JSON does not match the schema.
+    """
     if(type(el) is list):
-        return list(map(toJSON, el))
-    if(hasattr(el, 'tagName') and hasattr(el, 'attributes')):
-        return {
+        json_el = list(map(partial(toJSON, schema=None), el))
```

Comment:
Should this `partial` accept the current schema, `schema=schema`?

Reply:
no… this is to shortcut the recursive processing rather than introducing `_toJSON`.

Reply:
Also…I just realised, so long as our default argument is `None` we don't even need to do this!  because the recursive call was already setting schema to None since it was calling it without setting it!

Reply:
Silly me! Right, thanks for bearing with me and explaining. 👌